### PR TITLE
Modify SubmitBlock to SubmitBlocks in DA interace

### PR DESCRIPTION
**Pull request by Manav-Aggarwal**
## Overview

Resolves: #1080, #1084 

Adding DA Testing PR WIP: #1086 

## Checklist

- [x] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing - 1086 adds this
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

File: da/test/da_test.go [hunk 1]
```diff
@@ -124,7 +124,7 @@ func doTestRetrieve(t *testing.T, dalc da.DataAvailabilityLayerClient) {
 
 	for i := uint64(0); i < 100; i++ {
 		b := getRandomBlock(i, rand.Int()%20) //nolint:gosec
-		resp := dalc.SubmitBlock(ctx, b)
+		resp := dalc.SubmitBlocks(ctx, []*types.Block{b})
```
Comment (gupadhyaya) [inline da/test/da_test.go:127]: can you add a test that submits multiple blocks at same da height and tests retrieve?

**Review by gupadhyaya — COMMENTED** [review]
generally looks good. just needs an additional test for multiple block submission and retrieve imo.

File: da/test/da_test.go (hunk 1 above)
Comment (Manav-Aggarwal) [reply da/test/da_test.go:127]: There's an open PR for it that I needs debugging: https://github.com/rollkit/rollkit/pull/1085

**Review by tzdybal — APPROVED** [review]
Looks good. I think that as follow up issue we can think about improving error messages (because there's no way to know which block failed).

**Review by gupadhyaya — APPROVED** [review]

**Pull request merged**